frontend/src/metabase/query_builder/components/DataSelector/DataSelectorFieldPicker/DataSelectorFieldPicker.tsx

```diff
@@ -109,7 +109,7 @@ function renderItemWrapper(content: ReactNode) {
 
 const Header = ({ onBack, selectedTable }: HeaderProps) => (
   <HeaderContainer onClick={onBack}>
-    <Icon name="chevronleft" size={18} />
+    {onBack && <Icon name="chevronleft" size={18} />}
```

Comment:
`HeaderName` has `margin-left` - it should not be there when the back button icon is not shown.

Reply:
Fixed